spec/requests/cloud_networks_spec.rb

```diff
@@ -97,4 +97,35 @@
       expect(response).to have_http_status(:ok)
     end
   end
+
+  describe 'OPTIONS /api/cloud_networks' do
+    it 'returns a DDF schema for add when available via OPTIONS' do
+      zone = FactoryBot.create(:zone, :name => "api_zone")
+      provider = FactoryBot.create(:ems_network, :zone => zone)
+
+      allow(provider.class::CloudNetwork).to receive(:params_for_create).and_return('foo')
+
+      options(api_cloud_networks_url, :params => {:ems_id => provider.id})
+      options("#{api_cloud_networks_url}?ems_id=#{provider.id}")
```

Comment:
This feels like it should be 2 separate specs as opposed to calling one after the other.  I don't think the expects on lines 111-112 will pick up on anything set in the first call (or put another way, if you delete line 108, this test will still pass, so it's not actually being tested)

Reply:
I just tested it as 2 different specs, the spec with just line 108 failed. Do I keep it as 2 specs or leave it as 1 spec but remove line 108?

Reply:
I would keep it as 2 specs, and then we have to figure out why the first line fails.  If this was a copy past, we'll have to figure out why this isn't working.

Offhand, I'm wondering if it's a String vs Symbol thing (i.e. `:ems_id` vs `"ems_id"`), since that's the key you are looking for, but I thought `params` handled that.

Reply:
@Fryguy I just checked the cloud_subnets_spec.rb which is where this line initially came from, 
`    options(api_cloud_subnets_url, :params => {:ems_id => provider.id})
      options("#{api_cloud_subnets_url}?ems_id=#{provider.id}")`
After repeating the same process of removing the second options call and only testing the first the test case fails, so I am wondering if possibly the options call is being used incorrectly. I also checked the string vs symbol issue and it did not fix the problem.

Reply:
Interesting.  @agrare maybe you have some idea?

Reply:
I have no idea why they call `options` twice but only check the response once

Reply:
welp this is interesting

```
     def options
+      raise
       if params.key?("ems_id")
         options_by_ems_id
       elsif params.key?("id")
diff --git a/spec/requests/cloud_networks_spec.rb b/spec/requests/cloud_networks_spec.rb
index b0633efa..7387ec4f 100644
--- a/spec/requests/cloud_networks_spec.rb
+++ b/spec/requests/cloud_networks_spec.rb
@@ -106,7 +106,7 @@ RSpec.describe 'Cloud Networks API' do
       allow(provider.class::CloudNetwork).to receive(:params_for_create).and_return('foo')
 
       options(api_cloud_networks_url, :params => {:ems_id => provider.id})
-      options("#{api_cloud_networks_url}?ems_id=#{provider.id}")
+      #options("#{api_cloud_networks_url}?ems_id=#{provider.id}")
```

That first one isn't invoking the overridden options call at all?

Reply:
```
response.parsed_body
{"error"=>{"kind"=>"bad_request", "message"=>"809: unexpected token at 'ems_id=5000000000028'", "klass"=>"JSON::ParserError", 
```

Reply:
@agrare I have been trying out different ways of calling this and none seem to work:

`options(api_cloud_networks_url, :params => {:ems_id => provider.id})`

This line is returning a 400 error, but when I remove the ems_id and leave params as empty (:params => {}) then it returns a 200 response. So is the problem with the ems_id?

Reply:
What's wrong with `options("#{api_cloud_networks_url}?ems_id=#{provider.id}")`

Reply:
That one works, but @Fryguy wanted me to split up the 2 options call into separate specs so one calls this as well:
`options(api_cloud_networks_url, :params => {:ems_id => provider.id})`

Reply:
I'm wondering if that is even valid, I don't know where that came from or if it is expected to work.

Maybe a question for someone deeper into our API than me.

Reply:
I saw it in cloud_subnets_spec.rb so I used that as a starting point for my test cases. When I tried to put it as a separate  test case it failed and then I went back to cloud_subnets_spec.rb and tried the test case with only this options call and saw the same result of the test case failing. So I am wondering if we are either using `options(api_cloud_networks_url, :params => {:ems_id => provider.id})` incorrectly or if it is simply not expected to work.

Reply:
> So I am wondering if we are either using options(api_cloud_networks_url, :params => {:ems_id => provider.id}) incorrectly or if it is simply not expected to work.

Yes that's exactly what I'm wondering as well

Reply:
@Fryguy would it be possible to leave it as just 1 test case with the working api call:
`options("#{api_cloud_networks_url}?ems_id=#{provider.id}")`?

Reply:
Yeah, for now let's leave as just 1 working test case, and then can you open up a separate GitHub issue for the other one that we can remember to go back and investigate later?

Reply:
I made the change to the spec file and will open up the issue now.